testsuite/tests/typing-local/unregion.ml

```diff
@@ -0,0 +1,74 @@
+(* TEST
+ * expect *)
+
+(* TYPING TESTS *)
+(* TYPING TESTS *)
+(* TYPING TESTS *)
+(* the following two tests show that y is regional *)
+let foo x =
+  [%unregion] (
+    let local_ y = None in
+    (* y is not global *)
+    ref y
+  )
+[%%expect{|
+Line 5, characters 8-9:
+5 |     ref y
+            ^
+Error: This value escapes its region
+|}]
+
+let foo x =
+  [%unregion] (
+    let local_ y = Some x in
+    (* y is not local either *)
+    (* because it can escape *)
```

Comment:
This comment is a bit confusing. `y` is local it's just part of the caller's region. It would probably be better to say that `y` can escape the function even though it is local.